test/api/unit/authorizers/build.test.js

```diff
@@ -1,89 +1,107 @@
-    })
+      })
+        .then(({ user, build }) => authorizer.findOne(user, build))
+        .then(() => {
+          done(new Error('Expected authorization error'));
+        })
+        .catch((err) => {
+          expect(err).to.equal(403);
+          done();
+        })
+        .catch(done);
+    });
 
-    it("should reject if the build is not associated with one of the user's site even if the user started the build", done => {
-      const user = factory.user()
-      const build = factory.build({ user: user, site: factory.site() })
+    it("should reject if the build is not associated with one of the user's site even if the user started the build", (done) => {
+      const userProm = factory.user();
+      const buildProm = factory.build({ user: userProm, site: factory.site() });
 
-      Promise.props({ user, build }).then(({ user, build }) => {
-        return authorizer.findOne(user, build)
-      }).then(() => {
-        done(new Error("Expected authorization error"))
-      }).catch(err => {
-        expect(err).to.equal(403)
-        done()
-      }).catch(done)
-    })
-  })
+      Promise.props({ user: userProm, build: buildProm })
+        .then(({ user, build }) => authorizer.findOne(user, build)).then(() => {
+          done(new Error('Expected authorization error'));
+        })
+        .catch((err) => {
+          expect(err).to.equal(403);
+          done();
+        })
+        .catch(done);
+    });
+  });
 
-  describe("create(user, params)", () => {
-    it("should resolve if the build is associated with one of the user's site", done => {
-      const user = factory.user()
-      const site = factory.site({ users: Promise.all([user]) })
+  describe('create(user, params)', () => {
+    it("should resolve if the build is associated with one of the user's site", (done) => {
+      const userProm = factory.user();
+      const siteProm = factory.site({ users: Promise.all([userProm]) });
 
-      Promise.props({ user, site }).then(({ user, site }) => {
-        return authorizer.create(user, { user: user.id, site: site.id })
-      }).then(() => {
-        done()
-      }).catch(done)
-    })
+      Promise.props({ user: userProm, site: siteProm })
+        .then(({ user, site }) =>
+          authorizer.create(user, { user: user.id, site: site.id })
+        )
+        .then(() => {
+          done();
+        })
+        .catch(done);
+    });
 
-    it("should reject if the build is not associated with one of the user's sites", done => {
-      Promise.props({
-        user: factory.user(),
-        site: factory.site(),
-      }).then(({ user, build }) => {
-        return authorizer.create(user, { user: user.id, site: site.id })
-      }).then(() => {
-        done(new Error("Expected authorization error"))
-      }).catch(err => {
-        expect(err).to.equal(403)
-        done()
-      }).catch(done)
-    })
+    it("should reject if the build is not associated with one of the user's sites", (done) => {
```

Comment:
This is the test whose logic has been amended. It wasn't previously making a site that did belong to the user, so it wasn't really running through the logic of the target method.